src/core/analysis-context.ts

```diff
@@ -58,10 +61,15 @@ export class AnalysisContext {
   private _parsers = new Map<string, Parser<ParsedDocument<any, any>>>([
     ['html', new HtmlParser()],
     ['js', new JavaScriptParser()],
+    ['ts', new TypeScriptPreparser()],
     ['css', new CssParser()],
     ['json', new JsonParser()],
   ]);
 
+  private _languageAnalyzers = new Map<string, LanguageAnalyzer<any>>([
+    ['ts', new TypeScriptAnalyzer(this)],
```

Comment:
Should we use the same object to analyze both ts and js files? I suspect the answer is yes